### DISCUSSION TITLE
UI Tweaks

### DISCUSSION THREAD

**Discussion by flflow** [oobabooga/text-generation-webui#1143] [General]
Wanted show you my chat ui tweaks. What you think? It's raw, it's not ideal, but it's working and using less space. These are the first steps towards something better.
![Screenshot_2023-04-13-22-53-26-917_com android chrome](https://user-images.githubusercontent.com/32948829/231817114-2213189e-8a1f-46b8-8e44-e60dedfb0c10.jpg)
![2023-04-13 22 52 21](https://user-images.githubusercontent.com/32948829/231817125-7372fa12-85c3-4f7d-8c39-b006c16b7d46.png)

**Comment by MaiJames** [comment]
I guess the top is mobile and bottom is tablet, isn't it?

Overall, I think the design is looking good, but I have a few suggestions that could improve both the UI and UX:

UI Design:
- The prompt text box should have the same border radius as the rest of the UI for consistency.
- On mobile, the margins of the top part (conversation, prompt text box, and buttons) should match those of the bottom part.
- The arrow on the "Generate" button is too thin, and I recommend using a shade of the theme color instead of yellow. 

UX Design:
- Regarding the blue border of the text box, I suggest copying the approach used by Github in their text forms. They subtly change the background to a lighter shade and all borders to blue to indicate focus.
- Icons should be accompanied by descriptive text, except when the icon is very obvious. For instance, I couldn't tell what the other two icons do besides the one with the square and the arrow one, and part of that could be because I've been using the webui quite a lot. But the UI should be descriptive enough to be intuitive for a first time user. If there's enough space (and in this case, there is plenty), the use of the text labels would help users better understand the purpose of each button.
- Additionally, I noticed there is a "more" section hiding buttons. In my opinion, these buttons should be shown instead of hidden, as it could be confusing for users to discover them under the "more" section.

I hope this feedback is helpful :)

**Comment by Samjack1533** [comment]
@flflow Can you help me?
The model I use usually writes the contexts in ** ** or ( ) rather than * *. The UI doesn't italicize them automatically. Do you know how I can fix it?

**Comment by flflow** [comment]
Thank you for your reply.
Phone on top, PC version on bottom. I really dislike the current clutter of buttons in the original webui. 

What should be aligned, and not haphazardly, is all clear, the current version does not pretend to be something great.
I hide the buttons behind "more" because they are not needed so often, but on the screen of a mobile phone they takes up too much space. 
Icons on buttons made by emoji and utf symbols,its experimental and most simple. In the future, I plan to move the send text button to the right of the input field and I already have a blank for replacing the background of the buttons with images. I think, this will make the interface even less cluttered.

If anything, I am not related to the developers of this webui, I am the same user as anyone. I made these edits at my own discretion, it's so convenient for me. You have to carefully and little by little change something, because, often there are changes in webui that change the code very much, and after updating webui I manually apply my changes through "Diff Merge".

**Comment by xanthousm** [comment]
Using images for the buttons and hiding the less used ones definitely removes some of the clutter. Figuring out the best icon for each button will need some thinking though. Some of the new options like "send dummy message" are a bit abstract haha. 

I also made some personal UI tweaks (tabs stay on top of page, extensions in a separate tab) and hope that someone who has more experience with gradio and css makes the UI code in server.py more modular/safer from updates. Does your custom_css extension work without needing to change server.py?

**Reply by flflow** [reply]
I also had a feeling that the tabs should always be visible, but I still don’t use them so often. :D
Not exactly, css embedding on gradio level , but server.py had to be modified to assign a div id to buttons and important elements, also i changed order of buttons and hide some buttons with 
"with gr.Accordion("more...", open=False):"

**Comment by PaulCoddington** [comment]
The out-of-the-box Chat UI has a problem where the input box is always scrolled off the bottom of the screen while reading replies, or the replies are always scrolled off the top of the screen while typing input. It would be a welcome improvement to not to have to constantly scroll up and down with the scroll bar.

A light theme option (similar in style to GitHub default) would also be a welcome addition.
